plugins/generator-1.21.1/neoforge-1.21.1/templates/elementinits/menus.java.ftl

```diff
@@ -45,6 +45,51 @@ public class ${JavaModName}Menus {
 		= REGISTRY.register("${gui.getModElement().getRegistryName()}", () -> IMenuTypeExtension.create(${gui.getModElement().getName()}Menu::new));
 	</#list>
 
+	public interface MenuAccessor {
+	    HashMap<String, Object> getMenuState();
+	}
+
+	public static void updateMenuState(Player entity, int elementType, String name, Object elementState) {
+	    if (entity.containerMenu instanceof MenuAccessor menu) {
+	        HashMap<String, Object> menuState = menu.getMenuState();
+	        if (elementType == 0) {
+	            menuState.put("textfield:" + name, elementState);
+	        }
+	        if (elementType == 1) {
+            	menuState.put("checkbox:" + name, elementState);
+            }
+        }
+	}
+
+	public static void sendMenuStateUpdate(Player entity, int elementType, String name, Object elementState) {
+	    /*
+	     * There should be a synchronization code here to send data to the opposite side.
+	     */
+	    updateMenuState(entity, elementType, name, elementState); //This method will also be called from the network packet on the opposite side.
+	    if (entity.level().isClientSide) {
+	        ${JavaModName}Screens.onMenuStateUpdate(elementType, name, elementState); //A temporary method for updating GUI elements, after adding synchronization
+	        //it will be called from the packet received by the client from the server.
+	    }
+	}
+
+    <#-- At the moment this getter method returns a value only from the called side, it is not synchronized with the opposite side. -->
+	public static <T> T getMenuState(Entity entity, int elementType, String name, T defaultValue) {
+        if (entity instanceof Player _entity && _entity.containerMenu instanceof MenuAccessor accessor) {
+            HashMap<String, Object> menuState = accessor.getMenuState();
+            try {
+                 if (elementType == 0) {

```

Comment:
Replace ifs with `elementType + ":" + name`, then you can also omit `HashMap<String, Object> menuState = accessor.getMenuState();` and convert this to one line statement inside try block